pkg/k8s/api_test.go

```diff
@@ -4,26 +4,86 @@ import (
 	"testing"
 
 	"github.com/runconduit/conduit/pkg/shell"
+	"fmt"
+	"net/url"
 )
 
+type TestFixture struct{
+	testInput string
+	expectedOutput string
+}
 func TestKubernetesApiUrlFor(t *testing.T) {
+	const namespace = "some-namespace"
+	const extraPath = "/some/extra/path"
+	testData := []TestFixture{
+		{
+			testInput: "https://35.184.231.31",
+			expectedOutput:  fmt.Sprintf("https://35.184.231.31/api/v1/namespaces/%s%s", namespace, extraPath),
+		},
+		{
+			testInput: "35.184.231.31",
+			expectedOutput:  fmt.Sprintf("https://35.184.231.31/api/v1/namespaces/%s%s", namespace, extraPath),
+		},
+		{
+			testInput: "http://35.184.231.31",
+			expectedOutput:  fmt.Sprintf("http://35.184.231.31/api/v1/namespaces/%s%s", namespace, extraPath),
+		},
+	}
+
 	t.Run("Returns URL from base URL overridden in construction", func(t *testing.T) {
```

Comment:
I'd move the fixtures to inside the test case/`func` as they belong to it. Also, maybe the test name is a bit misleading now as this test is also testing the prepending of `http`?

Reply:
This is an interesting point, I am now essentially testing two different things in one test which I think violates the principles of unit testing. Do you think adding another test would suffice or would that be overkill?